components/exercises/chapter-4-oscar/Utils.tsx

```diff
@@ -7,34 +7,67 @@
  * @throws An error if step is zero.
  */
 export const getRangeWithStep = (
-    start: number,
-    end: number,
-    step: number = start < end ? 1 : -1
-  ): number[] => {
-    if (step === 0) {
-      throw new Error('Step cannot be zero.');
-    }
-  
-    const array = [];
-  
-    if (step > 0) {
-      for (let i = start; i <= end; i += step) array.push(i);
-    } else {
-      for (let i = start; i >= end; i += step) array.push(i);
-    }
-  
-    return array;
-  };
-  
-
-  /**
+  start: number,
+  end: number,
+  step: number = start < end ? 1 : -1
+): number[] => {
+  if (step === 0) {
+    throw new Error('Step cannot be zero.');
+  }
+
+  const array = [];
+
+  if (step > 0) {
+    for (let i = start; i <= end; i += step) array.push(i);
+  } else {
+    for (let i = start; i >= end; i += step) array.push(i);
+  }
+
+  return array;
+};
+
+/**
  * Calculates the sum of all numbers in an array.
  * @param arr - The array of numbers to sum.
  * @returns The sum of all numbers in the array.
  */
 export const arraySum = (arr: number[]): number => {
-    let sum = 0;
-    for (let i = 0; i < arr.length; i++) sum += arr[i];
-    return sum;
-  };
-  
\ No newline at end of file
+  let sum = 0;
+  for (let i = 0; i < arr.length; i++) sum += arr[i];
+  return sum;
+};
+
+/**
+ * Reverses the order of elements in an array.
+ * @param arr - The array to be reversed.
+ * @returns A new array with the elements in reverse order.
+ */
+export function getReverseArray(arr: any[]): any[] {
+  const newArr = [];
+
+  // loop through the array backwards and arr.length - 1 = 4 to match with indexation
```

Comment:
This comment doesn't make sense to me, better description could be:  
```js
// loop through the passed array in inverse order and add each item to the new array
```

Reply:
refactored